toolchain/check/impl_lookup.cpp

```diff
@@ -184,31 +184,47 @@ static auto FindAndDiagnoseImplLookupCycle(
 }
 
 struct InterfacesFromConstantId {
-  llvm::SmallVector<SemIR::SpecificInterface> interfaces;
+  llvm::ArrayRef<SemIR::SpecificInterface> interfaces;
   SemIR::BuiltinConstraintMask builtin_constraint_mask;
   bool other_requirements;
+  bool error = false;
+
+  static auto Error() -> InterfacesFromConstantId {
+    return {.interfaces = {},
+            .builtin_constraint_mask = SemIR::BuiltinConstraintMask::None,
+            .other_requirements = false,
+            .error = true};
+  }
 };
 
 // Gets the set of `SpecificInterface`s that are required by a facet type
 // (as a constant value), and any special requirements.
 static auto GetInterfacesFromConstantId(
-    Context& context, SemIR::ConstantId query_facet_type_const_id)
-    -> InterfacesFromConstantId {
+    Context& context, SemIR::LocId loc_id,
+    SemIR::ConstantId query_facet_type_const_id) -> InterfacesFromConstantId {
   auto facet_type_inst_id =
       context.constant_values().GetInstId(query_facet_type_const_id);
   auto facet_type_inst =
       context.insts().GetAs<SemIR::FacetType>(facet_type_inst_id);
   const auto& facet_type_info =
       context.facet_types().Get(facet_type_inst.facet_type_id);
-  auto identified_id = RequireIdentifiedFacetType(context, facet_type_inst);
-  auto interfaces_array_ref =
-      context.identified_facet_types().Get(identified_id).required_interfaces();
-  // Returns a copy to avoid use-after-free when the identified_facet_types
-  // store resizes.
-  return {
-      .interfaces = {interfaces_array_ref.begin(), interfaces_array_ref.end()},
-      .builtin_constraint_mask = facet_type_info.builtin_constraint_mask,
-      .other_requirements = facet_type_info.other_requirements};
+  // TODO: Get the complete facet type here.
+  auto identified_id =
+      RequireIdentifiedFacetType(context, facet_type_inst, [&] {
+        CARBON_DIAGNOSTIC(ImplLookupInIncompleteFacetType, Error,
+                          "impl lookup with facet for incomplete type {0}",
```

Comment:
This sounds more like "here's what was happening" (impl lookup with facet) than "here's what the problem was" (incomplete type...? is that right?).

Also why "facet for incomplete type" instead of "incomplete facet type"? Is there a nuance there I'm missing?

Maybe:

```suggestion
                          "facet type {0} is incomplete",
```

Reply:
Done